core/trino-spi/src/main/java/io/trino/spi/SplitWeight.java

```diff
@@ -87,7 +88,8 @@ public static SplitWeight fromProportion(double weight)
             throw new IllegalArgumentException("Invalid weight: " + weight);
         }
         // Must round up to avoid small weights rounding to 0
-        return fromRawValue((long) Math.ceil(weight * UNIT_VALUE));
+        long rawValue = (long) Math.ceil(weight * UNIT_VALUE);
+        return fromRawValue(Math.max(rawValue, MINIMUM_SPLIT_VALUE));
```

Comment:
I'm not sure we need to enforce this invariant in the `trino-spi` definition of a `SplitWeight`. While I didn't notice an improvement by going lower than this value in my test environment for Hive (which is probably comparable for Iceberg?), it's less clear that it still holds for connectors in general. I suspect, for example- that JDBC queries could have different characteristics and would compute the weight based on different heuristics.

Reply:
Yep, agreed. Made this change before reading your other comment, I will undo it.